lib/Service/IMipService.php

```diff
@@ -135,7 +135,11 @@ public function process(): void {
 				$sender = $imapMessage->getFrom()->first()->getEmail();
 				$recipient = $account->getEmail();
 				foreach ($imapMessage->scheduling as $schedulingInfo) { // an IMAP message could contain more than one iMIP object
-					if ($schedulingInfo['method'] === 'REPLY') {
+					if ($schedulingInfo['method'] === 'REQUEST') {
+						$processed = $this->calendarManager->handleIMipRequest($principalUri, $sender, $recipient, $schedulingInfo['contents']);
```

Comment:
What Psalm says. Open the code in PHPStorm. It will help you see that the interface is lacking the method.

Ref https://github.com/nextcloud/server/pull/47826#pullrequestreview-2288477381.

The app works with 30 and 31. Only 31 has the new API -> needs a method_exists check and a fallback

Reply:
> see that the interface is lacking the method.
Correct me if I am wrong but is this not expected since the server PR that the mail app needs is NOT merged yet? The handleIMipRequest is a need method that I added to the server (see required PR link in description) and is NOT merged yet.
 
> The app works with 30 and 31. Only 31 has the new API -> needs a method_exists check and a fallback
Okay, i was going to ask if we are going to back port this, but this answers my question.

Reply:
Regardless of the backport you can have deployments of Mail on older versions of Nextcloud where the method doesn't exist. So your code has to expect the method to be missing and have a workaround.